### PR TITLE
Add vscode debug support

### PR DESCRIPTION
**Pull request by enyaxu**
1. First run **watch** task.  **Terminal->Run Build Task…** chose **Build Boostnote** task. Or you can run `yarn run watch` command in terminal.
2. Now you can debug use **F5**

### CODE REVIEW

**Comment by ZeroX-DG** [comment]
I don't know why but `webpack-dev-server` return 404 for the bundle file `main.js`. I'm on Linux. Can you check again?

**Comment by enyaxu** [comment]
@ZeroX-DG I change **devServer** to **detail** mode. Can you check again?

**Comment by ZeroX-DG** [comment]
The webpack problem is somehow gone, but vscode now said that my breakpoint is `Unverified breakpoint`.

**Comment by enyaxu** [comment]
@ZeroX-DG When after start **Build Boostnote** task, You need to run debug config that named **BoostNote All**. And then you have two debug process running **Boostnote main** and **Boostnote renderer**. Now you need to switch process in vscode to debug your code. Like if you want to debug the code running in **Boostnote renderer**, you need to switch process to this.

This is because every **Electron** app have two process,  **Main** process and **Render** process.

**Comment by ZeroX-DG** [comment]
OK @enyaxu, it works now. I'm not familiar with vscode debug system so thank you for explaining. By the way, can you add the debug instruction to our docs? it's located here: https://github.com/BoostIO/Boostnote/blob/master/docs/debug.md

**Comment by enyaxu** [comment]
@ZeroX-DG Yes, I added the **debug instruction**  to **debug.md** doc.

File: docs/debug.md [hunk 1]
```diff
@@ -1,22 +1,36 @@
-# How to debug Boostnote (Electron app)
-This page is also available in [Japanese](https://github.com/BoostIO/Boostnote/blob/master/docs/jp/debug.md), [Korean](https://github.com/BoostIO/Boostnote/blob/master/docs/ko/debug.md), [Russain](https://github.com/BoostIO/Boostnote/blob/master/docs/ru/debug.md), [Simplified Chinese](https://github.com/BoostIO/Boostnote/blob/master/docs/zh_CN/debug.md), [French](https://github.com/BoostIO/Boostnote/blob/master/docs/fr/debug.md) and [German](https://github.com/BoostIO/Boostnote/blob/master/docs/de/debug.md).
+# How to debug BoostNote (Electron app)
```
Comment (ZeroX-DG) [inline docs/debug.md]: Please change `BoostNote` to `Boostnote` (the `n` is lowercase)

File: docs/debug.md [hunk 2]
```diff
@@ -1,22 +1,36 @@
...
 
+This page is also available in [Japanese](#)(https://github.com/BoostIO/Boostnote/blob/master/docs/jp/debug.md), [Korean](#)(https://github.com/BoostIO/Boostnote/blob/master/docs/ko/debug.md), [Russain](#)(https://github.com/BoostIO/Boostnote/blob/master/docs/ru/debug.md), [Simplified Chinese](#)(https://github.com/BoostIO/Boostnote/blob/master/docs/zh_CN/debug.md), [French](#)(https://github.com/BoostIO/Boostnote/blob/master/docs/fr/debug.md) and [German](#)(https://github.com/BoostIO/Boostnote/blob/master/docs/de/debug.md).
```
Comment (ZeroX-DG) [inline docs/debug.md]: Please remove `(#)` at the end of each link.

File: docs/debug.md [hunk 3]
```diff
@@ -1,22 +1,36 @@
...
+![Developer_Tools](#)(https://cloud.githubusercontent.com/assets/11307908/24343545/eff9f3a6-127b-11e7-94cf-cb67bfda634a.png)
 
 When errors occur, the error messages are displayed at the `console`.
 
-## Debugging
+### Debugging
 For example, you can use the `debugger` to set a breakpoint in the code like this:
 
-![debugger](https://cloud.githubusercontent.com/assets/11307908/24343879/9459efea-127d-11e7-9943-f60bf7f66d4a.png)
+![debugger](#)(https://cloud.githubusercontent.com/assets/11307908/24343879/9459efea-127d-11e7-9943-f60bf7f66d4a.png)
```
Comment (ZeroX-DG) [inline docs/debug.md]:
```suggestion
![debugger](https://cloud.githubusercontent.com/assets/11307908/24343879/9459efea-127d-11e7-9943-f60bf7f66d4a.png)
```

File: docs/debug.md [hunk 4]
```diff
@@ -1,22 +1,36 @@
...
 This is just an illustrative example, you should find a way to debug which fits your style.
 
-## References
+### References
 * [Official document of Google Chrome about debugging](https://developer.chrome.com/devtools)
+
+## Debug with Visual Studio Code
+
+1. Install **[Debugger for Chrome](https://marketplace.visualstudio.com/items?itemName=msjsdiag.debugger-for-chrome "Install Debugger for Chrome")** plugin for Visual Studio Code. Then restart it.
+2. Pressing **Shift+Command+B** or running ** Run Build Task** from the global **Terminal** menu, then pick the task named **Build Boostnote**.
```
Comment (ZeroX-DG) [inline docs/debug.md]:
```suggestion
2. Pressing **Shift+Command+B** or running **Run Build Task** from the global **Terminal** menu, then pick the task named **Build Boostnote**. Or run `yarn run watch` from the terminal.
```

File: docs/debug.md [hunk 5]
```diff
@@ -1,22 +1,36 @@
...
+3. When above task is running, open **Debug view** in **Activity Bar** on the side of VS Code or use shortcut **Shift+Command+D**.
+4. Select the configuration named **Boostnote All** from the **Debug configuration**, then click the green arrow button or press **F5**.
```
Comment (ZeroX-DG) [inline docs/debug.md]:
```suggestion
4. Select the configuration named **Boostnote All** from the **Debug configuration**, then click the green arrow button or press **F5** to start debugging.
```

File: docs/debug.md [hunk 6]
```diff
@@ -1,22 +1,36 @@
...
+5. Now you should find **Boostnote** is running. You will see two process is running, one named **Boostnote Main**,other named **Boostnote Renderer**. Now you can set **debug breakpoint** in vscode. If you find **breakpoints** is unverified, you need to switch process.
```
Comment (ZeroX-DG) [inline docs/debug.md]:
```suggestion
5. Now you should find **Boostnote** is running. You will see two processes running, one named **Boostnote Main** and the other named **Boostnote Renderer**. Now you can set **debug breakpoints** in vscode. If you find your **breakpoints** is unverified, you need to switch to the appropriate process between **Boostnote Renderer** and **Boostnote Main**.
```

**Pull request merged**